kotlinpoet/src/test/java/com/squareup/kotlinpoet/TypeSpecTest.kt

```diff
@@ -5336,6 +5336,83 @@ class TypeSpecTest {
     )
   }
 
+  // https://github.com/square/kotlinpoet/issues/1548
+  @Test fun overrideInternalAbstractFunctionVisibility() {
+    val baseClass = TypeSpec.classBuilder("Base")
+      .addModifiers(PUBLIC, ABSTRACT)
+      .addFunction(
+        FunSpec.builder("foo")
+          .addModifiers(INTERNAL, ABSTRACT)
+          .build(),
+      )
+      .build()
+    assertThat(baseClass.toString()).isEqualTo(
+      """
+      |public abstract class Base {
+      |  internal abstract fun foo(): kotlin.Unit
+      |}
+      |
+      """.trimMargin(),
+    )
+    val bassClassName = ClassName("", "Base")
+    val exampleClass = TypeSpec.classBuilder("Example")
+      .addModifiers(PUBLIC)
+      .superclass(bassClassName)
+      .addFunction(
+        FunSpec.builder("foo")
+          .addModifiers(KModifier.OVERRIDE)
+          .build(),
+      )
+      .build()
+    assertThat(exampleClass.toString()).isEqualTo(
+      """
+      |public class Example : Base() {
+      |  override fun foo(): kotlin.Unit {
+      |  }
+      |}
+      |
+      """.trimMargin(),
+    )
+    val example2Class = TypeSpec.classBuilder("Example2")
+      .addModifiers(PUBLIC)
+      .superclass(bassClassName)
+      .addFunction(
+        FunSpec.builder("foo")
+          .addModifiers(PUBLIC, KModifier.OVERRIDE)
+          .build(),
+      )
+      .build()
+    // Don't omit the public modifier here. The bytecode is different.
+    assertThat(example2Class.toString()).isEqualTo(
+      """
+      |public class Example2 : Base() {
+      |  public override fun foo(): kotlin.Unit {
+      |  }
+      |}
+      |
+      """.trimMargin(),
+    )
+    val example3Class = TypeSpec.classBuilder("Example3")
+      .addModifiers(INTERNAL)
+      .superclass(bassClassName)
+      .addFunction(
+        FunSpec.builder("foo")
+          .addModifiers(PUBLIC, KModifier.OVERRIDE)
+          .build(),
+      )
+      .build()
+    // We cannot omit the public modifier here. The bytecode is different.
```

Comment:
Not sure I understand what "The bytecode is different" means in this context.

Reply:
Here is what it means. Generated by `Kotlin Bytecode` from IntelliJ IDEA

![image](https://github.com/square/kotlinpoet/assets/14316223/760238a6-80a9-4f7f-b9d9-3ba1a08cf8c1)

![image](https://github.com/square/kotlinpoet/assets/14316223/608007b0-48e7-4510-8cc1-f1edbe3498f4)

Reply:
Got it. If the intent is to explain why the "public" modifier can't be omitted, I don't think the reference to bytecode is very helpful. I'll provide suggestions on how to reword this, but please let me know if I misunderstood your intent.